lib/collection.js

```diff
@@ -0,0 +1,291 @@
+var Events = require('./events'),
+    Collection,
+    ARRAY_PROXY_METHODS = ['forEach', 'some', 'every', 'filter', 'map', 'reduce'];
+
+function clone(obj) {
```

Comment:
`Object.create(proto)` creates new object with prototype `proto`. It doesn't clone. 
````javascript
var o = {a: 'a'};
var o2 = Object.create(o);
o.a = 'b';
console.log(o2.a) // 'b';
````
Here is live demo https://jsfiddle.net/Ltesu6ry/
I think threre is not we are looking for.